extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnPageReaderImpl.java

```diff
@@ -264,16 +267,61 @@ private static Encoding getEncoding(final org.apache.parquet.format.Encoding enc
         return org.apache.parquet.column.Encoding.valueOf(encoding.name());
     }
 
-    private int readRowCountFromPageV1(final DataPageV1 page) {
+    /**
+     * Reads a little-endian integer from the provided {@link DataInput}.
+     *
+     * @throws IOException If an error occurs while reading from the input stream
+     * @implNote DataInput reads big-endian, so we need to reverse the bytes
+     */
+    private static int readLittleEndianInt(final DataInput in) throws IOException {
+        return Integer.reverseBytes(in.readInt());
+    }
+
+    /**
+     * Helper struct for holding a ByteBuffer.
+     */
+    // TODO Better name
+    private static class BytesArrayHolder implements SafeCloseable {
+        private byte[] bytes;
+
+        BytesArrayHolder() {}
+
+        private void setBytes(final byte[] bytes) {
+            this.bytes = bytes;
+        }
+
+        @Override
+        public void close() {}
+    }
+
+    /**
+     * Get a cached byte array from the channel context, or create a new one if it doesn't exist or is too small.
+     */
+    private static byte[] getCachedBuffer(
+            @NotNull final SeekableChannelContext channelContext,
+            final int size) {
+        final BytesArrayHolder bufferHolder =
+                (BytesArrayHolder) channelContext.apply(BUFFER_HOLDER_KEY, BytesArrayHolder::new);
+        if (bufferHolder.bytes == null || bufferHolder.bytes.length < size) {
+            bufferHolder.setBytes(new byte[size]);
+            // TODO This will allocate a new buffer every time we see a slightly larger page than before. We could
+            // potentially cache the next power of 2 buffer size, hoping that should be enough for most pages.
```

Comment:
I would certainly do something more efficient than "always allocate the minimum".
Devin used `io.deephaven.base.MathUtil#roundUpArraySize` recently for the json-jackson stuff.
One thing to consider (although it's OK if you don't) is to interact with chunk pooling... maybe use `io.deephaven.chunk.sized.SizedByteChunk` to hold a chunk (which holds a byte array), instead of your `ByteArrayHolder`.